build/compiler.py

```diff
@@ -222,6 +222,59 @@ def generate(self, force=False):
     return True
 
 
+class TypescriptGenerator(object):
+  def __init__(self, source_files, build_name):
+    self.source_files = _canonicalize_source_files(source_files)
+    self.output = _get_source_path('dist/' + build_name + '.d.ts')
+
+  def generate(self, force=False):
+    """Generates TypeScript definitions for the files in |self.source_files|.
+
+    Args:
+      force: Generate the output even if the inputs have not changed.
+
+    Returns:
+      True on success; False on failure.
+    """
+    if not force and not _must_build(self.output, self.source_files):
+      return True
+
+    ts_generator = _get_source_path('build/typescript/main.ts')
+    ts_node = _get_source_path('node_modules/ts-node/dist/bin')
+    tsconfig = _get_source_path('build/typescript/tsconfig.json')
+
+    cmd_line = ['node', ts_node, '-P', tsconfig, ts_generator, '--output', self.output]
+    cmd_line += self.source_files
+
+    if shakaBuildHelpers.execute_get_code(cmd_line) != 0:
+      logging.error('TypeScript generation failed')
+      return False
+
+    return True
+  
+
+class TypescriptCompiler(object):
+  def __init__(self, project):
+    self.project = _get_source_path(project)
+
+  def compile(self):
+    """Compiles the TypeScript project defined by the tsconfig in |self.project|.
+
+    Returns:
+      True on success; False on failure.
+    """
+
+    tsc = _get_source_path('node_modules/typescript/lib/tsc')
+    cmd_line = ['node', tsc, '-p', self.project]
```

Comment:
Same here, please use our node utility function to build the correct command-line.